### ISSUE TITLE
Documentation for 3D scattering output

### ISSUE THREAD

**Issue by mburhanpurkar** [kymatio/kymatio#639]
I am hoping to use the 3D scattering function, but I am somewhat confused by its output. 

I saw the documentation (https://www.kymat.io/userguide.html#id14) for output size, but when I ran the HarmonicScattering3D function with J=7, L=4, P=1 (I assume this is the length of the integral_powers argument?) on a (2, 128, 128, 128) batch, the output had shape (2, 36, 5, 3), which I believe is not consistent with what is listed in the documentation. 

Could you clarify what the dimensions of the output of the 3D scattering function correspond to (or point me to documentation that specifies this)? Also, what is C meant to represent in that section of the documentation?

System info:
Darwin-19.6.0-x86_64-i386-64bit
Python 3.7.7 (default, Mar 26 2020, 10:32:53) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
NumPy 1.18.1
SciPy 1.4.1
PyTorch 1.4.0
Kymatio 0.2.0

**Comment by eickenberg** [comment]
Thanks for raising this issue.

Just to understand correctly: You gave the `HarmonicScattering3D` object an argument `integral_powers=list_of_integral_powers`, where the length of this list is 1 instead of the default `integral_powers=(0.5, 1., 2.)`?

In this case the result is indeed wrongly shaped.

We have an upcoming sprint in February during which we will definitely look into this. In the mean time, would you be able to write a minimal example piece of code that exhibits this discrepancy? Something like this for example:

```python
from kymatio.torch import HarmonicScattering3D
J = 7
L = 4
integral_powers = (1.,)
scattering = HarmonicScattering3d(shape=(64, 64, 64), J=J, L=L, integral_powers=integral_powers)
x = torch.randn(2, 64, 64, 64)
out = scattering(x)
assert out.shape == x.shape[0], J * (J - 1) // 2, L + 1, len(integral_powers)
```

^ I wrote the above without running it, so it may be inexact. The goal is to have a failing test case that we can then fix.

**Comment by mburhanpurkar** [comment]
That's right. Here is an example:
```python
from kymatio import HarmonicScattering3D
from kymatio.scattering3d.backend.numpy_backend import backend
import numpy as np
D = 128
integral_powers = (1.,)
for J in range(1, 7):
    for L in range(1, 4):
        d = np.random.randn(2, D, D, D)
        S = HarmonicScattering3D(J=J, shape=(D, D, D), L=L, sigma_0=1., max_order=2, backend=backend,
                                 frontend='numpy', rotation_covariant=True) 
        order_0 = backend.compute_integrals(d, integral_powers)
        coeffs = S(d)
        print(L, J, np.shape(coeffs), (d.shape[0], J * (J - 1) // 2, L + 1, len(integral_powers)))
```
Gives the output:
```
J   L   shape(coeffs)      doc reference
-----------------------------------
1   1   (2, 3, 2, 3)       (2, 0, 2, 1)
2   1   (2, 3, 3, 3)       (2, 0, 3, 1)
3   1   (2, 3, 4, 3)       (2, 0, 4, 1)
1   2   (2, 6, 2, 3)       (2, 1, 2, 1)
2   2   (2, 6, 3, 3)       (2, 1, 3, 1)
3   2   (2, 6, 4, 3)       (2, 1, 4, 1)
1   3   (2, 10, 2, 3)      (2, 3, 2, 1)
2   3   (2, 10, 3, 3)      (2, 3, 3, 1)
3   3   (2, 10, 4, 3)      (2, 3, 4, 1)
1   4   (2, 15, 2, 3)      (2, 6, 2, 1)
2   4   (2, 15, 3, 3)      (2, 6, 3, 1)
3   4   (2, 15, 4, 3)      (2, 6, 4, 1)
1   5   (2, 21, 2, 3)      (2, 10, 2, 1)
2   5   (2, 21, 3, 3)      (2, 10, 3, 1)
3   5   (2, 21, 4, 3)      (2, 10, 4, 1)
1   6   (2, 28, 2, 3)      (2, 15, 2, 1)
2   6   (2, 28, 3, 3)      (2, 15, 3, 1)
3   6   (2, 28, 4, 3)      (2, 15, 4, 1)
```
By the way, the documentation [here](https://www.kymat.io/userguide.html#id14) is inconsistent with output shape `x.shape[0], J * (J - 1) // 2, L + 1, len(integral_powers)`.

It would also be really helpful if the documentation for the output in general for the 3D scattering transform could be expanded. The 2D API has an `out_type="list"` option that helps with this, but for the 3D documentation, I'm not certain what "concatenated along the feature axis" [here](https://www.kymat.io/codereference.html) means for the ordering of the coefficients.

Thanks!

**Comment by janden** [comment]
Closed by #820.
